src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -1032,4 +1047,235 @@ private StringConcatFactory() {
         // no instantiation
     }
 
+    /**
+     * Bytecode StringBuilder strategy.
+     *
+     * <p>This strategy emits StringBuilder chains as similar as possible
+     * to what javac would. No exact sizing of parameters or estimates.
+     */
+    private static final class SimpleStringBuilderStrategy {
+        static final int CLASSFILE_VERSION = 52; // JDK 8
```

Comment:
Alternatively, this can use `ClassFileFormatVersion.latest().major()` which exists since JDK 20.

Reply:
As the original code just had 52 verbatim and I don't know how far we'll want to backport this I think it's fine to leave this at 52.